src/Microsoft.TemplateEngine.Core.Contracts/IConditionedConfigurationElement.cs

```diff
@@ -1,16 +1,33 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#nullable enable
+
 using Microsoft.TemplateEngine.Abstractions;
 
 namespace Microsoft.TemplateEngine.Core.Contracts
 {
     public interface IConditionedConfigurationElement
     {
-        string Condition { get; }
+        /// <summary>
+        /// Gets or sets the condition string to be evaluated.
+        /// </summary>
+        string? Condition { get; }
 
+        /// <summary>
+        /// Gets the cached result of the evaluation.
+        /// Make sure <see cref="EvaluateCondition(IEngineEnvironmentSettings, IVariableCollection)"/> is called
```

Comment:
We can't automatically call it in the property, because it requires arguments that we don't have access to. But I agree with you on how this looks bad: we are just trying to access the result without knowing if the values was really evaluated or not and we don't have the option to evaluate it right there because we don't have access to the required objects to pass them as arguments.

It sounds like it would be better to have a new type like `EvaluatedConditionedConfiguration` or something that we can pass around once we do the evaluation.